File: leetcode/mergeAlternately.go
package main

import "fmt"

func mergeAlternately(word1 string, word2 string) string {
    mergedStr := ""

	wordOneLength := len(word1)
	wordTwoLength := len(word2)

	i := 0

	for ;i < wordOneLength; i++ {
		mergedStr += string(word1[i])
		if (i < wordTwoLength) {
			mergedStr += string(word2[i])
		} else {
			break
		}
	}

	if (i+1 < wordOneLength) {
		mergedStr += word1[i+1:wordOneLength];
	}
    if (i < wordTwoLength) {
        mergedStr += word2[i+1:wordTwoLength];
    }

	return mergedStr
}

func main () {
	answer1 := mergeAlternately("abc", "pqr"); // apbqcr
	// answer2 := mergeAlternately("ab", "pqrs"); // apbqrs
	// answer3 := mergeAlternately("abcd", "pq"); // apbqcd
	
	fmt.Println(answer1);
}

File: leetcode/mergeAlternately.js
function mergeAlternately(word1, word2) {

    let mergedStr = "";
    const wordOneLength = word1.length
    const wordTwoLength = word2.length;
    let i = 0;
    for (; i < wordOneLength; i++) {
        mergedStr += word1[i];
        if (i < wordTwoLength) {
            mergedStr += word2[i]
        } else {
            break;
        }
    }

    if (i+1 < wordOneLength) {
        mergedStr += word1.slice(i+1, wordOneLength);
    } 

    if (i < wordTwoLength) {
        mergedStr += word2.slice(i, wordTwoLength)
    }

    return mergedStr;
}

const answer1 = mergeAlternately("abc", "pqr"); // apbqcr
const answer2 = mergeAlternately("ab", "pqrs"); // apbqrs
const answer3 = mergeAlternately("abcd", "pq"); // apbqcd

console.log(answer1);
console.log(answer2);
console.log(answer3);